sdk/core/Azure.Core/src/RequestContent.cs

```diff
@@ -79,6 +79,14 @@ public abstract class RequestContent : IDisposable
         /// <returns>An instance of <see cref="RequestContent"/> that wraps a <see cref="DynamicData"/>.</returns>
         public static RequestContent Create(DynamicData content) => new DynamicDataContent(content);
 
+        /// <summary>
+        /// Generates JSON for JSON Merge Patch updates corresponding to the changes made to the DynamicData instance.
+        /// </summary>
+        /// <param name="content"></param>
+        /// <returns></returns>
+        /// <exception cref="NotImplementedException"></exception>
+        public static RequestContent CreatePatch(DynamicData content) => new DynamicDataContent(content, 'P');
```

Comment:
This isn't strictly needed for the convenience layer, but if we're enabling it at the convenience layer, it would make sense to enable it at the protocol layer as well, and this is needed for that.